tensorflow-core/tensorflow-core-generator/src/main/java/org/tensorflow/op/generator/AttributeType.java

```diff
@@ -0,0 +1,31 @@
+/*
+  Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================
+ */
+package org.tensorflow.op.generator;
+
+public enum AttributeType {
+  String, Int, Float, Bool, Shape, Tensor, Type;
```

Comment:
All caps for the enum constants?

Reply:
I'm using the names to resolve the methods, so I wanted to avoid having a redundant constructor parameter.  I can do that if it's a big deal.

Reply:
You can put the parameter in the enum and get it back out of the object. I think directly using the enum constant names is probably a bad idea.